snowsql: add page (#9519)
tldr-pages/tldr: pages/common/snowsql.md

# snowsql

> SnowSQL command-line client for Snowflake's Data Cloud.
> More information: <https://docs.snowflake.com/en/user-guide/snowsql.html>.

> - Connect to an instance at {{account}}.snowflakecomputing.com (password can be provided in prompt or configuration file):

Reviewer: ```suggestion
- Connect to a specific instance at account.snowflakecomputing.com (password can be provided in prompt or configuration file):
```

- This change is optional - no corresponding style guide requirement exist.
- The corresponding change should be applied to `.pl` translation if it's used here.
- We often prefer use `a specific` statement.
- There is no requirement to enclose referred to placeholders in braces in descriptions.

> - Connect to a specific instance at account.snowflakecomputing.com (password can be provided in prompt or configuration file):

Reviewer 2: Based on https://github.com/tldr-pages/tldr/blob/main/pages/common/firebase.md
```suggestion
- Connect to a specific instance at <https://account.snowflakecomputing.com> (password can be provided in prompt or configuration file):
```

`snowsql --accountname {{account}} --username {{username}} --dbname {{database}} --schemaname {{schema}}`

> - Connect to an instance specified by the configuration file (by default, `~/.snowsql/config`):

Reviewer: ```suggestion
- Connect to an instance specified by a specific configuration file (by default, `~/.snowsql/config`):
```

Almost the same issue like I've described above.

> - Connect to an instance specified by a specific configuration file (by default, `~/.snowsql/config`):

Reviewer 2: ```suggestion
- Connect to an instance specified by a specific configuration file (defaults to `~/.snowsql/config`):
```

> `snowsql --config {{.snowsql/config}}`

Reviewer: ```suggestion
`snowsql --config {{~/.snowsql/config}}`
```

- This change is optional - no corresponding style guide requirement exist.
- The corresponding change should be applied to `.pl` translation if it's used here.
- It's better to have identical paths.

> `snowsql --config {{~/.snowsql/config}}`

Reviewer 2: If we are using a non-default file:
```suggestion
`snowsql --config {{path/to/configuration_file}}`
```
Also, we try to avoid abbreviations

Author: Done, also changed in the translation.

> - Connect to the default instance, specifying a MFA token:

Reviewer 2: Not sure about `MFA`, I think readers will Google for `MFA` :thinking: . What if ...?
```suggestion
- Connect to the default instance using a token for multi-factor authentication:
```
I based this suggestion using https://docs.snowflake.com/en/user-guide/snowsql-start.html#connection-syntax.

 @kbdharun wdyt?

Reviewer 2: @Holek ?

Reviewer 3: > Not sure about `MFA`, I think readers will Google for `MFA` :thinking: . What if ...?
> ```suggestion
> - Connect to the default instance using a token for multi-factor authentication:
> ```
> I based this suggestion using https://docs.snowflake.com/en/user-guide/snowsql-start.html#connection-syntax.
> 
>  @kbdharun wdyt?

Agreed, It would be nice to specify it for new users.

Author: Sure, what's funny is that I explained that as such in Polish translation already.

`snowsql --mfa-passcode {{token}}`

- Execute a single SQL query or SnowSQL command on the default connection (useful in shell scripts):

`snowsql --query '{{query}}'`

> - Execute commands from a file on the default connection:

Reviewer: ```suggestion
- Execute commands from a specific file on the default connection:
```

- This change is optional - no corresponding style guide requirement exist.
- The corresponding change should be applied to `.pl` translation if it's used here.
- We often prefer use `a specific` statement.
- There is no requirement to enclose referred to placeholders in braces in descriptions.

> `snowsql --filename {{file.sql}}`

Reviewer 3: ```suggestion
`snowsql --filename {{path/to/file.sql}}`
```
